src/Jackett/Services/IndexerManagerService.cs

```diff
@@ -25,70 +24,152 @@ public interface IIndexerManagerService
         IEnumerable<IIndexer> GetAllIndexers();
 
         void InitIndexers();
-        void InitCardigannIndexers(string path);
+        void InitCardigannIndexers(IEnumerable<string> path);
         void InitAggregateIndexer();
-        void SortIndexers();
     }
 
     public class IndexerManagerService : IIndexerManagerService
     {
-        private IContainer container;
-        private Logger logger;
-        private Dictionary<string, IIndexer> indexers = new Dictionary<string, IIndexer>();
         private ICacheService cacheService;
         private IIndexerConfigurationService configService;
+        private IProtectionService protectionService;
+        private IWebClient webClient;
+
+        private Logger logger;
+
+        private Dictionary<string, IIndexer> indexers = new Dictionary<string, IIndexer>();
         private AggregateIndexer aggregateIndexer;
 
-        public IndexerManagerService(IContainer c, IIndexerConfigurationService config, Logger l, ICacheService cache)
+        public IndexerManagerService(IIndexerConfigurationService config, IProtectionService protectionService, IWebClient webClient, Logger l, ICacheService cache)
         {
-            container = c;
             configService = config;
+            this.protectionService = protectionService;
+            this.webClient = webClient;
             logger = l;
             cacheService = cache;
         }
 
         public void InitIndexers()
         {
-            logger.Info("Using HTTP Client: " + container.Resolve<IWebClient>().GetType().Name);
-
-            foreach (var idx in container.Resolve<IEnumerable<IIndexer>>().OrderBy(_ => _.DisplayName))
+            logger.Info("Using HTTP Client: " + webClient.GetType().Name);
+
+            var ixs = new IIndexer[]{
```

Comment:
I was toying with that idea, however there's no guarantee that all of the indexer's constructor arguments are the same. They mostly use the same services now (an exception would be AnimeTorrents, which does not accept IWebClient), but I cannot guarantee that there won't be a next implementation that changes that. In fact, one of the next things I would like to do is to revisit the dependencies of every indexer and see whether they are in fact required or not, which might change certain constructor parameter list.

I might be missing context here as I don't know whether the code-based implementation or the cardigann definition file-based one is preferred, if one or the other is obsoleted, than this is probably safe to be changed. However if there's no preference there, I wouldn't limit any new implementation to using the exact same parameter list just to spare a couple lines of code.